pkg/controller/plan/kubevirt.go

```diff
@@ -555,11 +572,136 @@ func (r *KubeVirt) getPVCs(vm *plan.VMStatus) (pvcs []core.PersistentVolumeClaim
 		pvcAnn := pvc.GetAnnotations()
 		if pvcAnn[kVM] == vmLabels[kVM] && pvcAnn[kPlan] == vmLabels[kPlan] {
 			pvcs = append(pvcs, *pvc)
+		} else {
+			if r.isOvirtProvider(vm) {
+				ovirtVm := &ovirt.Workload{}
+				err = r.Source.Inventory.Find(ovirtVm, vm.Ref)
+				if err != nil {
+					return
+				}
+				for _, da := range ovirtVm.DiskAttachments {
+					if da.Disk.ID == pvc.Spec.DataSource.Name {
+						pvcs = append(pvcs, *pvc)
+						break
+					}
+				}
+			}
 		}
 	}
 	return
 }
 
+func (r *KubeVirt) createVolumes(vm ref.Ref) (err error) {
+	ovirtVm := &ovirt.Workload{}
+	err = r.Source.Inventory.Find(ovirtVm, vm)
+	if err != nil {
+		return
+	}
+	sourceUrl, err := url.Parse(r.Source.Provider.Spec.URL)
+	if err != nil {
+		return
+	}
+
+	storageName := &r.Context.Map.Storage.Spec.Map[0].Destination.StorageClass
+	for _, da := range ovirtVm.DiskAttachments {
+		populatorCr := r.OvirtVolumePopulator(da, sourceUrl)
+		failure := r.Client.Create(context.Background(), populatorCr, &client.CreateOptions{})
+		if failure != nil && !k8serr.IsAlreadyExists(failure) {
+			return failure
+		}
+
+		accessModes, volumeMode, failure := r.getStorageProfileModes(*storageName)
+		if failure != nil {
+			return failure
+		}
+
+		pvc := r.Builder.PersistentVolumeClaimWithSourceRef(da, storageName, populatorCr.Name, accessModes, volumeMode)
+		err = r.Client.Create(context.TODO(), pvc, &client.CreateOptions{})
+		if err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+// Build an OvirtVolumePopulator for XDiskAttachment and source URL
+func (r *KubeVirt) OvirtVolumePopulator(da ovirt.XDiskAttachment, sourceUrl *url.URL) *v1beta1.OvirtVolumePopulator {
+	return &v1beta1.OvirtVolumePopulator{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      da.DiskAttachment.ID,
+			Namespace: r.Plan.Spec.TargetNamespace,
+		},
+		Spec: v1beta1.OvirtVolumePopulatorSpec{
+			EngineURL:        fmt.Sprintf("https://%s", sourceUrl.Host),
+			EngineSecretName: r.Source.Secret.Name,
+			DiskID:           da.Disk.ID,
+		},
+	}
+}
+
+func (r *KubeVirt) getOvirtPVCs(vm ref.Ref, step *plan.Step) (ready bool, err error) {
```

Comment:
why the name is getOvirtPVCs? it doesn't seem to return PVCs here

Reply:
changed, let me know if it fits better for you

Reply:
it was renamed to arePVCsReady which is better but I think the name should reflect that it only handles PVCs for oVirt